File: internal/db/team.go
package db

import (
	"shinobi_showdown/internal/game"

	"github.com/google/uuid"
)

type ActorConfig struct {
	AbilityID *uuid.UUID             `json:"ability_ID"`
	ActionIDs []uuid.UUID            `json:"action_IDs"`
	Focus     *game.ActorFocus       `json:"focus"`
	ItemID    *uuid.UUID             `json:"item_ID"`
	AuxStats  map[game.ActorStat]int `json:"aux_stats"`
}

type TeamActor struct {
	ActorID uuid.UUID   `json:"actor_ID"`
	Config  ActorConfig `json:"config"`
}

type TeamConfig struct {
	Actors []TeamActor `json:"actors"`
	Name   string      `json:"name"`
}

File: web/src/lib/game/team.ts
import type { TeamActor } from "../stores/config";
import type { ActorDef } from "./actor";

function makeConfigFromDef(def: ActorDef): TeamActor {
  return {
    actor_ID: def.actor_ID,
    config: {
      ability_ID: def.abilities[0]?.ID ?? null,
      item_ID: null,
      action_IDs: [],
      focus: 'none',
      aux_stats: {
        hp: 0,
        stamina: 0,
        speed: 0,
        attack: 0,
        defense: 0,
        chakra_attack: 0,
        chakra_defense: 0,
      },
    },
  }
}

export { makeConfigFromDef }
